doc/developers/contributing.rst

```diff
@@ -268,100 +249,115 @@ If any of the above seems like magic to you, then look up the `Git documentation
 <http://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html>`_ on the
 web, or ask a friend or another contributor for help.
 
-If some conflicts arise between your branch and the ``master`` branch, you need
-to merge ``master``. For that, you first need to fetch the ``upstream``, and
-then merge its ``master`` into your branch::
+In order to keep your local branch synchronized with the latest
+changes of the main scikit-learn repository, you will need to add
+the ``upstream`` remote::
 
-  $ git fetch upstream
-  $ git merge upstream/master
+    $ git remote add upstream https://github.com/scikit-learn/scikit-learn.git
 
-Subsequently, you need to solve the conflicts. You can refer to the `Git
-documentation related to resolving merge conflict using the command line
-<https://help.github.com/articles/resolving-a-merge-conflict-using-the-command-line/>`_.
+You can now synchronize your local branch with::
 
-.. note::
+    $ git fetch upstream
+    $ git merge upstream/master
 
-   In the past, the policy to resolve conflicts was to rebase your branch on
-   ``master``. GitHub interface deals with merging ``master`` better than in
-   the past.
+Subsequently, you might need to solve the conflicts. You can refer to the `Git
+documentation related to resolving merge conflict using the command line
+<https://help.github.com/articles/resolving-a-merge-conflict-using-the-command-line/>`_.
 
 
-Contributing pull requests
---------------------------
+Pull request checklist
+----------------------
```

Comment:
This list is essentially the same but I tried to prioritize tests (with some tips that were IMO missing), documentation and pep8.